src/dbnode/persist/fs/files.go

```diff
@@ -291,55 +292,128 @@ func timeAndIndexFromFileName(fname string, componentPosition int) (time.Time, i
 	return t, int(index), nil
 }
 
-type infoFileFn func(fname string, infoData []byte)
+type forEachInfoFileSelector struct {
+	fileSetType    persist.FileSetType
+	contentType    persist.FileSetContentType
+	filePathPrefix string
+	namespace      ident.ID
+	shard          uint32 // shard only applicable for data content type
+}
 
-func forEachInfoFile(filePathPrefix string, namespace ident.ID, shard uint32, readerBufferSize int, fn infoFileFn) {
+type infoFileFn func(fname string, id FileSetFileIdentifier, infoData []byte)
+
+func forEachInfoFile(
+	args forEachInfoFileSelector,
+	readerBufferSize int,
+	fn infoFileFn,
+) {
 	matched, err := filesetFiles(filesetFilesSelector{
-		fileSetType:    persist.FileSetFlushType,
-		contentType:    persist.FileSetDataContentType,
-		filePathPrefix: filePathPrefix,
-		namespace:      namespace,
-		shard:          shard,
+		fileSetType:    args.fileSetType,
+		contentType:    args.contentType,
+		filePathPrefix: args.filePathPrefix,
+		namespace:      args.namespace,
+		shard:          args.shard,
 		pattern:        infoFilePattern,
 	})
 	if err != nil {
 		return
 	}
 
-	shardDir := ShardDataDirPath(filePathPrefix, namespace, shard)
+	var dir string
+	switch args.fileSetType {
+	case persist.FileSetFlushType:
+		switch args.contentType {
+		case persist.FileSetDataContentType:
+			dir = ShardDataDirPath(args.filePathPrefix, args.namespace, args.shard)
+		case persist.FileSetIndexContentType:
+			dir = NamespaceIndexDataDirPath(args.filePathPrefix, args.namespace)
+		default:
+			return
```

Comment:
should these be errors/panics?

Reply:
As above, decided to currently leave it as its modeled right now.  Perhaps I should open an issue to make ForEachInfoFile... function families deal with this better?

Reply:
sounds good yea.

Reply:
Opened issue: